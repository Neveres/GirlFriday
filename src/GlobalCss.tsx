import React from 'react'
import { css, Global } from '@emotion/react'

export const GlobalCss = () => (
  <Global
    styles={css`
      * {
        font-family: ${fontFamily.ubuntu};
        font-style: normal;
        color: ${colors.textWhite};
      }

      body {
        margin: 0;
        background-color: ${colors.bgPrimary};
      }

      #root {
        min-height: 100vh;
        position: relative;
      }

      .mobile-page-title {
        font-weight: ${fontWeight.lighter};
        font-size: ${fontSize.xxLarge};
        line-height: 150%;
        margin-bottom: 24px;
      }
    `}
  />
)

export const fontSize = {
  xxxLarge: '48px',
  xxLarge: '24px',
  xLarge: '16px',
  large: '14px',
  medium: '13px',
  small: '12px',
  xSmall: '11px',
}

export const fontWeight = {
  bolder: 700,
  bold: 600,
  lighter: 400,
}

export const fontFamily = {
  ubuntu: 'Ubuntu',
}

export const colors = {
  bgPrimary: '#181818',
  bgSecondary: '#1B1B1B',
  textWhite: '#ffffff',
  textGray: '#b2b2b2',
}
