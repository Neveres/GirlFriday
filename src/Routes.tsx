import React, { lazy } from 'react'
import { Navigate, useRoutes } from 'react-router-dom'

const Home = lazy(() => import('src/containers/Home'))
const SearchResult = lazy(() => import('src/containers/SearchResult'))
const Tags = lazy(() => import('src/containers/Tags'))

export enum PagePath {
  Root = '/',
  Home = '/home',
  SearchResult = '/search-result',
  Tags = '/tags',
}

export const menuItems: GirlFriday.NavBarItem[] = [
  {
    name: 'Home',
    path: PagePath.Home,
    isInNavBar: true,
    withSideBar: true,
    withNavBar: true,
  },
  {
    name: 'Search Result',
    path: PagePath.SearchResult,
    isInNavBar: false,
    withSideBar: true,
    withNavBar: false,
  },
  {
    name: 'Tags',
    path: PagePath.Tags,
    isInNavBar: true,
    withSideBar: false,
    withNavBar: false,
  },
]

export const Routes = () =>
  useRoutes([
    {
      path: PagePath.Home,
      element: <Home />,
    },
    {
      path: PagePath.SearchResult,
      element: <SearchResult />,
    },
    {
      path: PagePath.Tags,
      element: <Tags />,
    },
    {
      path: '*',
      element: <Navigate to={PagePath.Home} replace />,
    },
  ])
