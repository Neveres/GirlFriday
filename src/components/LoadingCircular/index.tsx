export { default, default as LoadingCircular } from './LoadingCircular'
