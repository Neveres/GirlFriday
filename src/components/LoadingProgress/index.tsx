export { default, default as LoadingProgress } from './LoadingProgress'
