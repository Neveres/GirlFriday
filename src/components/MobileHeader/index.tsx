export { default, default as MobileHeader } from './MobileHeader'
