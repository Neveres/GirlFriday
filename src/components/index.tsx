export { NavBar } from './NavBar'
export { AppBody } from './AppBody'
export { AppContext } from './AppContext'
export { Input } from './Input'
export { Button } from './Button'
export { Slider } from './Slider'
