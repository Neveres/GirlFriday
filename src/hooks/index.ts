export * from './useInnerWidth'
export * from './useTags'
export * from './useUsers'
export * from './useStorage'
export * from './useFriends'
