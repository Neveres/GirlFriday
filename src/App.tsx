import React, { Suspense, useReducer, useMemo } from 'react'
import { useLocation } from 'react-router'
import { NavBar, AppBody, AppContext } from 'src/components'
import { Routes } from 'src/Routes'
import { useDevice } from 'src/hooks'
import { reducer, SET_MOBILE, SET_SEARCH_PARAMETERS } from 'src/libraries'
import { GlobalCss } from './GlobalCss'

const initialState = {
  isMobile: false,
  searchParameters: {
    page: 1,
    pageSize: 10,
    keyword: '',
  } as GirlFriday.SearchParameters,
}

const App = () => {
  const { pathname } = useLocation()

  const [state, dispatch] = useReducer(reducer, initialState)
  const actions = useMemo(
    () => ({
      setSearchParameters(searchParameters: GirlFriday.SearchParameters) {
        dispatch({ type: SET_SEARCH_PARAMETERS, ...state, searchParameters })
      },
      setMobile(isMobile: boolean) {
        dispatch({ type: SET_MOBILE, ...state, isMobile })
      },
    }),
    [state],
  )
  useDevice({ setMobile: actions.setMobile })
  const valueOfAppContext = useMemo(
    () => ({
      state,
      actions,
    }),
    [state, actions],
  )

  return (
    <>
      <AppContext.Provider value={valueOfAppContext}>
        <NavBar pathname={pathname} />

        <AppBody pathname={pathname}>
          <Suspense fallback={<div>Loading...</div>}>
            <Routes />
          </Suspense>
        </AppBody>
      </AppContext.Provider>

      <GlobalCss />
    </>
  )
}

export default App
