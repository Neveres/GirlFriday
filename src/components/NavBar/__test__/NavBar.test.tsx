import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { create } from 'react-test-renderer'
import { NavBar, AppContext } from 'src/components'
import { PagePath } from 'src/Routes'

describe('NavBar', () => {
  const paths = Object.values(PagePath).map((value) => value)

  test('should render well under desktop layout', () => {
    for (let i = 0, maxI = paths.length - 1; i < maxI; i++) {
      const testRenderer = create(
        <BrowserRouter>
          <AppContext.Provider
            value={
              {
                state: {
                  isMobileLayout: false,
                },
              } as any
            }
          >
            <NavBar pathname={paths[i]} />
          </AppContext.Provider>
        </BrowserRouter>,
      )
      expect(testRenderer.toJSON()).toMatchSnapshot()
    }
  })

  test('should render well under mobile layout', () => {
    for (let i = 0, maxI = paths.length - 1; i < maxI; i++) {
      const testRenderer = create(
        <BrowserRouter>
          <AppContext.Provider
            value={
              {
                state: {
                  isMobileLayout: true,
                },
              } as any
            }
          >
            <NavBar pathname={paths[i]} />
          </AppContext.Provider>
        </BrowserRouter>,
      )
      expect(testRenderer.toJSON()).toMatchSnapshot()
    }
  })
})
