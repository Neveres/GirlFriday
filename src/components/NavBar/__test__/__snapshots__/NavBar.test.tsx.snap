// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render well under desktop layout 1`] = `
<div
  className="css-3i2qem-NavBar"
>
  <div
    className="navBar-logo-container"
  >
    <div
      className="navBar-logo"
    >
      LOGO
    </div>
  </div>
  <div
    className="navBar-items"
  >
    <div
      className="navBar-item home "
    >
      <a
        href="/home"
        onClick={[Function]}
      >
        <img
          src="union_dark.svg"
        />
      </a>
      <span
        className="dot"
      />
    </div>
    <div
      className="navBar-item tags "
    >
      <a
        href="/tags"
        onClick={[Function]}
      >
        <img
          src="union_dark.svg"
        />
      </a>
      <span
        className="dot"
      />
    </div>
  </div>
</div>
`;

exports[`NavBar should render well under desktop layout 2`] = `
<div
  className="css-3i2qem-NavBar"
>
  <div
    className="navBar-logo-container"
  >
    <div
      className="navBar-logo"
    >
      LOGO
    </div>
  </div>
  <div
    className="navBar-items"
  >
    <div
      className="navBar-item home focused"
    >
      <a
        href="/home"
        onClick={[Function]}
      >
        <img
          src="union_light.svg"
        />
        Home
      </a>
    </div>
    <div
      className="navBar-item tags "
    >
      <a
        href="/tags"
        onClick={[Function]}
      >
        <img
          src="union_dark.svg"
        />
      </a>
      <span
        className="dot"
      />
    </div>
  </div>
</div>
`;

exports[`NavBar should render well under desktop layout 3`] = `
<div
  className="css-3i2qem-NavBar"
>
  <div
    className="navBar-logo-container"
  >
    <div
      className="navBar-logo"
    >
      LOGO
    </div>
  </div>
  <div
    className="navBar-items"
  >
    <div
      className="navBar-item home focused"
    >
      <a
        href="/home"
        onClick={[Function]}
      >
        <img
          src="union_light.svg"
        />
        Home
      </a>
    </div>
    <div
      className="navBar-item tags "
    >
      <a
        href="/tags"
        onClick={[Function]}
      >
        <img
          src="union_dark.svg"
        />
      </a>
      <span
        className="dot"
      />
    </div>
  </div>
</div>
`;

exports[`NavBar should render well under desktop layout 4`] = `
<div
  className="css-3i2qem-NavBar"
>
  <div
    className="navBar-logo-container"
  >
    <div
      className="navBar-logo"
    >
      LOGO
    </div>
  </div>
  <div
    className="navBar-items"
  >
    <div
      className="navBar-item home "
    >
      <a
        href="/home"
        onClick={[Function]}
      >
        <img
          src="union_dark.svg"
        />
      </a>
    </div>
    <div
      className="navBar-item tags focused"
    >
      <a
        href="/tags"
        onClick={[Function]}
      >
        <img
          src="union_light.svg"
        />
        Tags
      </a>
    </div>
  </div>
</div>
`;

exports[`NavBar should render well under mobile layout 1`] = `null`;

exports[`NavBar should render well under mobile layout 2`] = `
Array [
  <div
    className="css-11x547p-NavBar"
  >
    <div
      className="navBar-logo"
    >
      LOGO
    </div>
  </div>,
  <div
    className="css-l1k74l-NavBar"
  >
    <div
      className="navBar-items"
    >
      <div
        className="navBar-item home focused"
      >
        <a
          href="/home"
          onClick={[Function]}
        >
          <img
            src="union_light.svg"
          />
        </a>
      </div>
      <div
        className="navBar-item tags "
      >
        <a
          href="/tags"
          onClick={[Function]}
        >
          <img
            src="union_dark.svg"
          />
        </a>
        <span
          className="dot"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`NavBar should render well under mobile layout 3`] = `null`;

exports[`NavBar should render well under mobile layout 4`] = `null`;
